src/m365/spo/commands/listitem/listitem-roleinheritance-reset.ts

```diff
@@ -0,0 +1,107 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  id: number;
+  listId?: string;
+  listTitle?: string;
+}
+
+class SpoListItemRoleInHeritanceResetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.LISTITEM_ROLEINHERITANCE_RESET;
+  }
+
+  public get description(): string {
+    return 'Restores the role inheritance of list item, file, or folder';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== 'undefined';
+    telemetryProps.listTitle = typeof args.options.listTitle !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    let requestUrl: string = `${args.options.webUrl}/_api/web/lists`;
+
+    if (args.options.listId) {
+      requestUrl += `(guid'${encodeURIComponent(args.options.listId)}')`;
+    }
+    else {
+      requestUrl += `/getbytitle('${encodeURIComponent(args.options.listTitle as string)}')`;
+    }
+
+    const requestOptions: any = {
+      url: `${requestUrl}/items(${args.options.id})/resetroleinheritance`,
+      method: 'POST',
+      headers: {
+        'accept': 'application/json;odata=nometadata',
+        'content-type': 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .post(requestOptions)
+      .then(_ => cb(), (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--id <id>'
+      },
+      {
+        option: '--listId [listId]'
+      },
+      {
+        option: '--listTitle [listTitle]'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+    if (isValidSharePointUrl !== true) {
+      return isValidSharePointUrl;
+    }
+
+    if (!args.options.listId && !args.options.listTitle) {
+      return `Specify listId or listTitle`;
+    }
+
+    if (args.options.listId && args.options.listTitle) {
+      return `Specify listId or listTitle but not both`;
+    }
```

Comment:
one cool thing I learned lately is that we may use `optionSets` instead of this validation conditions. Maybe you could consider removing this and adding something like this instead 🤔👍

```
public optionSets(): string[][] | undefined {
    return [
      [ 'listId', 'listTitle' ]
    ];
  }
```

Reply:
Great suggestion @Adam-it. We should definitely use option sets in newly added commands.